core/repository/api/src/main/java/org/eclipse/rdf4j/repository/RepositoryConnection.java

```diff
@@ -622,7 +623,18 @@ public void exportStatements(Resource subj, IRI pred, Value obj, boolean include
 	 * @see #rollback()
 	 * @see #setIsolationLevel(IsolationLevel)
 	 */
-	public void begin(IsolationLevel level) throws RepositoryException;
+	void begin(IsolationLevel level) throws RepositoryException;
+
+	default void begin(TransactionSetting... settings) {
```

Comment:
Javadoc on this would be useful. Specifically the general contract and the default implementation behaviour of selecting the first `IsolationLevel` rather than using the last or another decision algorithm.

Reply:
Good point regarding javadoc. Fwiw at this point I would prefer _not_ to commit ourselves to a particular decision algorithm for this, and just say that "behavior is undefined when more than one isolation level is supplied".